### ISSUE TITLE
Specify network interface names instead of static IP addresses [13614]

### ISSUE THREAD

**Issue by fujitatomoya** [eProsima/Fast-DDS#1943]
## Feature Request

we'd like to open discussion against our use case, which is container with multiple network interface including container orchestration. whitelist https://fast-dds.docs.eprosima.com/en/latest/fastdds/transport/whitelist.html, it only supports static IP addresses but not with network interface name, since the NIC name almost is fixed but IP maybe dynamic assigned by DHCP or container orchestration framework. so the whitelist profile only support IP seems more static in some use cases.

**Comment by RFRIEDM-Trimble** [comment]
For IPv6, the RFC standards recommend using zone ID, which yea, I believe is the same as the NIC names. 
https://datatracker.ietf.org/doc/html/rfc4007

Boost's ipv6 address constructor uses the `scope_id` integer representation of the interface name. https://www.boost.org/doc/libs/1_66_0/doc/html/boost_asio/reference/ip__address_v6/address_v6/overload2.html

I can't tell if the constructor's here support sending scope or zone id. 
https://www.boost.org/doc/libs/1_66_0/doc/html/boost_asio/reference/ip__address_v6/make_address_v6/overload4.html

Regardless, whitelisting by interface address, in ipv6 land, does not appear to be the standard way the other networking API's and RFC's seem to indicate. If two interface adapters have the same ip address, then the current method of whitelisting that uses address binding for the socket does not appear to be deterministic. Interface names are.

**Comment by JLBuenoLopez-eProsima** [comment]
I am going to proceed and close this feature request as it is available in Fast DDS since release 2.4.0. Even though that release indicates that the feature is for defining Discovery Server locators, the truth is that it is possible to define any locator in the library by this method. I have issued eProsima/Fast-DDS-docs#319 to clarify this point.

**Comment by ciandonovan** [comment]
Do you mean you can specify the interface whitelist directly in the address field such as

```
<interfaceWhiteList>
    <address>lo</address>
</interfaceWhiteList>
``` 

or through using a locator with

```
<interfaceWhiteList>
    <locator>
        <udpv4>
            <address>lo</address>
        </udpv4>
    </locator>
</interfaceWhiteList>
```

or something else entirely?
